src/main/java/io/jenkins/plugins/gitlabbranchsource/helpers/GitLabHelper.java

```diff
@@ -16,14 +20,29 @@ public static GitLabApi apiBuilder(String serverName) {
         if (server != null) {
             PersonalAccessToken credentials = server.getCredentials();
             if (credentials != null) {
-                return new GitLabApi(server.getServerUrl(), credentials.getToken().getPlainText());
+                return new GitLabApi(server.getServerUrl(), credentials.getToken().getPlainText(), null, getProxyConfig());
             }
-            return new GitLabApi(server.getServerUrl(), GitLabServer.EMPTY_TOKEN);
+            return new GitLabApi(server.getServerUrl(), GitLabServer.EMPTY_TOKEN, null, getProxyConfig());
         }
         throw new IllegalStateException(
             String.format("No server found with the name: %s", serverName));
     }
 
+    public static Map<String, Object> getProxyConfig () {
```

Comment:
I think we should support https proxy settings..But i'm not sure

Reply:
It's a good point. In Jenkins plugin manager proxy configuration, serverUrl is a string without protocol (http or https), and there is no field to specify the protocol to use.
Based on this configuration, we are not able to determine if it is http or https protocol. We could presume that if it is 80, use HTTP and 443 use HTTPS, but proxy port can still be 8443 or something else.

I think if we want to support both http and https, this plugin must have it's own proxy user settings and proxy support implementation.

Reply:
Hello, is it good for you ?

Reply:
It is. I wish I would have write access here..meh

Reply:
I hope so !